### PR TITLE
rename, and doc-comment, and elide unused methods

### PR DESCRIPTION
**Pull request by zancas**
# Rationale:

Rename components of the TransactionRecord to emphasize how they are related to the TransactionRecord.

This significantly clarifies the code.

# Comments

Input (in this context) means "input of value to the TransactionRecord".

By highlighting that the Record is a Capability-Specific dataset, I hope further clarity is gained.

By marking the differential publicity of different pieces of data I hope to add clarity, but also to suggest an enhancement in the form of new language-level categorization of this data. 

Some of the elided methods might need to be re-added when the `pub` of the record is removed or diminished, but I think it's better to "burn that bridge when we come to it" rather than carry extra kruft.

### CODE REVIEW

**Comment by Oscar-Pepper** [comment]
NAck: comments in codeshare for discussion

File: zingolib/src/wallet/transaction_record.rs [hunk 1]
```diff
@@ -25,47 +25,67 @@ use crate::{
...
     /// Txid of this transaction. It's duplicated here (It is also the Key in the HashMap that points to this
-    /// WalletTx in LightWallet::txs)
+    /// TransactionRecord in LightWallet::TransactionRecordsById)
+    ///  # ChainPublic
     pub txid: TxId,
 
-    /// List of all nullifiers spent by this wallet in this Tx.
-    pub spent_sapling_nullifiers: Vec<sapling_crypto::Nullifier>,
+    /// List of all Sapling nullifiers used by this Capability to add value to the Sapling Value Pool.
+    ///  # Outgoing
```
Comment (Oscar-Pepper) [inline zingolib/src/wallet/transaction_record.rs:53]: lets have a discussion about this PR, i have left some comments in chat. Outgoing here highlights the type of confusion we might unleash by defining inputs as nullifiers, (and only nullifiers). nullifiers are actually public on chain so should fall into the category of public. but i suppose there is confusion here that the value comes from a note/output only seen by the "Outgoing" capability. however, a nullifier is public and a link to that output, not the output itself or the value it holds

**Review by Oscar-Pepper — CHANGES_REQUESTED** [review]
I think this PR needs discussion

File: zingolib/src/wallet/transaction_record.rs (hunk 1 above)
Comment (fluidvanadium) [reply zingolib/src/wallet/transaction_record.rs:53]: i like the clarity that these nullifiers are inputs to THIS transaction

**Review by fluidvanadium — DISMISSED** [review]
+4 for illustrative clear comments

File: zingolib/src/wallet/transaction_record.rs (hunk 1 above)
Comment (zancas) [reply zingolib/src/wallet/transaction_record.rs:53]: The 32-byte number (IIRC) nullifier value is public, but which output it nullifies is **critically** private.   So I agree that there's  room to improve these doc-comments, but I think we should merge and iterate on them.

Comment (Oscar-Pepper) [reply zingolib/src/wallet/transaction_record.rs:53]: sorry i dont agree with merging new code that is confusing/incorrect on the basis we can change it later. lets change it now. nullifiers are public and when we add an extra field to transaction records that stores a ref to the notes spent in this transaction we can call that field Outgoing

**Pull request closed without merging**